http-server/src/main/java/io/airlift/http/server/HttpServer.java

```diff
@@ -177,6 +178,9 @@ public HttpServer(
             baseHttpConfiguration.setResponseHeaderSize(toIntExact(config.getMaxResponseHeaderSize().toBytes()));
         }
 
+        // allow encoded slashes to occur in URI paths
+        baseHttpConfiguration.setUriCompliance(UriCompliance.LEGACY);
```

Comment:
I think we should use `UriCompliance.AMBIGUOUS_PATH_SEPARATOR`

Reply:
`UriCompliance.AMBIGUOUS_PATH_SEPARATOR` is a `Violation`. Only `UriCompliance` instances are accepted which is a `final` class. The only one that worked was `LEGACY`.